### PR TITLE
#1357 Respect markdown code boundaries for code execution

### PR DESCRIPTION
**Pull request by vineetgorhe**
https://github.com/jupyterlab/jupyterlab/issues/1357

Submitting my first PR to initiate discussion and a quick review of the approach.

### CODE REVIEW

File: packages/fileeditor-extension/src/index.ts [hunk 1]
```diff
@@ -174,6 +189,95 @@ function activate(app: JupyterLab, registry: IDocumentRegistry, restorer: ILayou
     return tracker.currentWidget !== null;
   }
 
+
+  class CodeSnippet {
```
Comment (ellisonbg) [inline packages/fileeditor-extension/src/index.ts]: Let's move the structures and functions that parse markdown into code blocks into a separate model in the coreutils package and add tests.

File: packages/fileeditor-extension/src/index.ts [hunk 2]
```diff
@@ -174,6 +189,95 @@ function activate(app: JupyterLab, registry: IDocumentRegistry, restorer: ILayou
...
+        return marker;
+      }
+    }
+    return '';
+  }
+  /**
+   * Construct all code snippets from current text
+   * (this could be potentially optimized if we can cache and detect differences)
+   */
+  function findCodeSnippets(): CodeSnippet[] {
```
Comment (ellisonbg) [inline packages/fileeditor-extension/src/index.ts]: I think we should have a standalone function that takes a markdown string and returns a list of code block objects. I think there are other places we might want to use this function.

File: packages/fileeditor-extension/src/index.ts (hunk 1 above)
Comment (vineetgorhe) [inline packages/fileeditor-extension/src/index.ts]: Will do.

File: packages/fileeditor-extension/src/index.ts (hunk 2 above)
Comment (vineetgorhe) [inline packages/fileeditor-extension/src/index.ts]: Will do.

**Comment by ellisonbg** [comment]
Hi @vineetgorhe I can start to work on a review and testing now. Do you have time to rebase?

**Comment by ellisonbg** [comment]
@vineetgorhe I will pick this one up

**Comment by vineetgorhe** [comment]
Hi - I might be able to pick this up later in the week.

**Comment by ellisonbg** [comment]
Closing in favor of rebased #2270

**Pull request closed without merging**